src/Akka.Streams.Kafka/Stages/Consumers/Abstract/SubSourceLogic.cs

```diff
@@ -365,14 +472,14 @@ private void EmitSubSourcesForPendingPartitions()
                     _pendingPartitions = _pendingPartitions.Remove(topicPartition);
                     _partitionsInStartup = _partitionsInStartup.Add(topicPartition);
 
-                    var subSourceStage = new SubSourceStreamStage(
+                    var subSourceStage = new SubSourceStage<K,V,TMessage>(
```

Comment:
Each time we are given a new partition, we must emit a new `SubSourceStage` that belongs to that partition.